modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ComponentMessageProcessorObjectFactory.java

```diff
@@ -82,7 +88,13 @@ public P doGetObject() {
                                                                              nestedChain)));
 
       // For MULE-18771 we need access to the chain's location to create a new event and sdk context
-      nestedChain.setAnnotations(this.getAnnotations());
+      // Update: For W-15158118, to avoid issues with the registration of the chain in the component locator, we are not
+      // adding the location as the location key but as another annotation.
+      Map<QName, Object> annotations = new HashMap<>(this.getAnnotations());
+      ComponentLocation chainLocation = ((DefaultComponentLocation) annotations.get(LOCATION_KEY));
+      annotations.remove(LOCATION_KEY);
```

Comment:
I think it's best for the chain to not return null when asked for its location. This can generate problesm down the line. I think the second annotation approach is fine, but the LOCATION_KEY should still be present and point to `{parentLocation}/processors` (as you had it before)